### PR TITLE
Fix multiple deposits

### PR DESCRIPTION
**Pull request by vaibhavchellani**

### CODE REVIEW

File: cmd/start.go [hunk 1]
```diff
@@ -97,7 +97,7 @@ func storeGenesisData(bz *bazooka.Bazooka, DBI *db.DB, cfg config.Configuration)
 	accountTreeDepth, err := bz.SC.AccountRegistry.DEPTH(&opts)
 	common.PanicIfError(err)
 
-	err = DBI.InitAccountTree(int(accountTreeDepth.Uint64()))
+	err = DBI.InitAccountTree(int(accountTreeDepth.Uint64()) + 1)
```
Comment (ChihChengLiang) [inline cmd/start.go:100]: Nice catch here! The DEPTH in the accountTree contract is misleading, created an issue to fix it later https://github.com/thehubbleproject/hubble-contracts/issues/457😓 .

File: aggregator/aggregator.go [hunk 2]
```diff
@@ -15,13 +15,14 @@ import (
...
-	ErrIncorrectTxCount = errors.New("inaccurate number of transactions")
+	AggregatingService       = "aggregator"
+	ErrIncorrectTxCount      = errors.New("inaccurate number of transactions")
+	ErrNotEnoughTransactions = errors.New("not enough transactions")
 )
 
 // Aggregator is the service which is supposed to create batches
 // It has the following tasks:
-// 1. Pick txs from the mempool
+// 1. Pick txs from the mempoolinaccurate number of transactione
```
Comment (ChihChengLiang) [inline aggregator/aggregator.go:25]: Is this accidentally committed?

File: core/batch.go [hunk 3]
```diff
@@ -235,61 +185,36 @@ type MassMigrationBody struct {
...
+	tokenIDBytes := math.U256Bytes(m.TokenID)
+	amountBytes := math.U256Bytes(m.Amount)
+	feeReceiverBytes := math.U256Bytes(m.FeeReceiver)
+	return encodePacked(accountRootBytes, signatureBytes, spokeIDBytes, withdrawRootBytes, tokenIDBytes, amountBytes, feeReceiverBytes, m.Txs), nil
+}
 
-	return ByteArray(Keccak256(data)), nil
+func encodePacked(input ...[]byte) []byte {
+	return bytes.Join(input, nil)
+}
```
Comment (ChihChengLiang) [inline core/batch.go:212]: Great work solving this!

File: core/merkleTree.go [hunk 4]
```diff
@@ -17,7 +17,7 @@ func init() {
 		panic(err)
 	}
 
-	DefaultHashes, err = GenDefaultHashes(100)
+	DefaultHashes, err = GenDefaultHashes(100, ZeroLeaf)
```
Comment (ChihChengLiang) [inline core/merkleTree.go:20]: No change is required here. Just want to point out that we call the zero/empty leaf differently between the contract and the client.
https://github.com/thehubbleproject/hubble-contracts/pull/376#issue-534377693

**Review by ChihChengLiang — COMMENTED** [review]
Awesome work on fixing the encodePacked issue!
This is the first pass of the review. Might take a second round again.

File: cmd/start.go (hunk 1 above)
Comment (vaibhavchellani) [reply cmd/start.go:100]: This one lead to finding another big issue, so no harm no foul.

File: aggregator/aggregator.go (hunk 2 above)
Comment (vaibhavchellani) [reply aggregator/aggregator.go:25]: yeah, will change

File: core/merkleTree.go [hunk 5]
```diff
@@ -36,14 +36,14 @@ type Tree struct {
 // NewTree creates a merkle tree from given leaves
 // the order of leaves will be preserved
 func NewTree(leaves []ByteArray) (newTree Tree, err error) {
+	if len(leaves)%2 != 0 {
+		leaves = append(leaves, ZeroLeaf)
+	}
```
Comment (ChihChengLiang) [inline core/merkleTree.go:41]: Here we handled the odd number of leaves only in the bottom level. But the odd number of nodes should be handled at every level.

For example, say we have 9 leaves to merklise. This makes a 4 level tree (not counting the root).
1. We add defaultHash[0] in level 0 leaves. After the ascend, level 1 has 5 nodes
2. We add defaultHash[1] in level 1 nodes to make them 6 nodes. After the ascend, level 2 has 3 nodes.
3. We add defaultHash[2] in level 2 nodes to make them 4 nodes. After the ascend, level 3 has 2 nodes.

I suggest this logic is better handled in the `ascend` method.

File: aggregator/aggregator.go [hunk 6]
```diff
@@ -159,10 +165,18 @@ func (a *Aggregator) processAndSubmitBatch(txs []core.Tx) {
 
 func (a *Aggregator) processTxs(txs []core.Tx) (commitments []core.Commitment, err error) {
 	var txsInCommitment []int
+
+	if len(txs) < int(a.cfg.TxsPerCommitment) {
+		return commitments, ErrNotEnoughTransactions
+	}
```
Comment (ChihChengLiang) [inline aggregator/aggregator.go:171]: Does a transaction take a long time to be submitted on chain if the commitment is unfilled?

**Review by ChihChengLiang — APPROVED** [review]
LGTM. The comment on merkle tree might deserve its own issue and PR to resolve.

File: aggregator/aggregator.go (hunk 6 above)
Comment (vaibhavchellani) [reply aggregator/aggregator.go:171]: not really, it just might be non economical to submit unfilled commitments on-chain for a coordinator, so we allow him to configure that tightly.

**Pull request merged**